### PR TITLE
Fixes #9556 - Better prompt for input on Password

### LINKED ISSUE #9556: Password Util does not ask for password
**Jetty version(s)**
10

**Java version/vendor** `(use: java -version)`
Adoptiom JRE 11

**OS type/version**
Windows

**Description**
Password utility is used to obfuscate passwords, e.g. for keystore.

When running the utility without any parameter 

D:\apps\_current\jre11\bin\java.exe -cp d:\apps\_current\jetty10\lib\jetty-util-10.0.13.jar   org.eclipse.jetty.util.security.Password

 prints out:

_Usage - java org.eclipse.jetty.util.security.Password [<user>] <password>
If the password is ?, the user will be prompted for the password_

**How to reproduce?**

Starting the programm with ?, \? or "?" produces always the same result:

_OBF:1e2n
MD5:D1457b72C3Fb323a2671125aEf3eAb5d_

But I was expecting some input prompt.

### PR DESCRIPTION
**Pull request by joakime**
Fixes how we prompt user for inputs on the `org.eclipse.jetty.util.security.Password` command line tool.

This PR (initially) shows the following results ...

_Usage text_

``` shell
$ java -cp target/jetty-util-10.0.15-SNAPSHOT.jar org.eclipse.jetty.util.security.Password 
Usage - java org.eclipse.jetty.util.security.Password [<user>] <password> --prompt
Argument options:
  org.eclipse.jetty.util.security.Password
     No arguments, will show this help
  org.eclipse.jetty.util.security.Password myusername
     username only, will prompt for entries
  org.eclipse.jetty.util.security.Password myusername ?
     username with question mark password, will prompt for entries
  org.eclipse.jetty.util.security.Password myusername secretpassword
     username with password, will produce obfuscation results
  org.eclipse.jetty.util.security.Password --prompt
     will prompt for entries
```

_Specifically asking for input prompts_

``` shell
$ java -cp target/jetty-util-10.0.15-SNAPSHOT.jar org.eclipse.jetty.util.security.Password --prompt
Username: user⏎ 
Password: pass⏎ 
OBF:1y0s1v1p1v2p1y0y
MD5:1a1dC91c907325C69271DdF0C944Bc72
CRYPT:usGhvISzyaLuU
```

_Username provided, no password, prompt user_

``` shell
$ java -cp target/jetty-util-10.0.15-SNAPSHOT.jar org.eclipse.jetty.util.security.Password ?
Username[?]: user⏎ 
Password: pass⏎ 
OBF:1y0s1v1p1v2p1y0y
MD5:1a1dC91c907325C69271DdF0C944Bc72
CRYPT:usGhvISzyaLuU

$ java -cp target/jetty-util-10.0.15-SNAPSHOT.jar org.eclipse.jetty.util.security.Password myusername
Username[myusername]: ⏎ 
Password: mypassword⏎ 
OBF:1uh41zly1x8g1vu11ym71ym71vv91x8e1zlk1ugm
MD5:34819d7bEeAbB9260a5c854bC85b3e44
CRYPT:..slnl7JMUqP.

$ java -cp target/jetty-util-10.0.15-SNAPSHOT.jar org.eclipse.jetty.util.security.Password myusername ?
Username[myusername]: ⏎
Password: mypassword⏎
OBF:1uh41zly1x8g1vu11ym71ym71vv91x8e1zlk1ugm
MD5:34819d7bEeAbB9260a5c854bC85b3e44
CRYPT:..slnl7JMUqP.
```

_Prompt for input, no username provided, output doesn't contain CRYPT_

``` shell
$ java -cp target/jetty-util-10.0.15-SNAPSHOT.jar org.eclipse.jetty.util.security.Password --prompt
Username: ⏎
Password: mypass⏎
OBF:1xfd1zt11uha1ugg1zsp1xfp
MD5:A029D0Df84Eb5549C641E04a9eF389E5

```

_Username and password provided as arguments, no prompt for input_

``` shell
$ java -cp target/jetty-util-10.0.15-SNAPSHOT.jar org.eclipse.jetty.util.security.Password user pass
OBF:1y0s1v1p1v2p1y0y
MD5:1a1dC91c907325C69271DdF0C944Bc72
CRYPT:usGhvISzyaLuU
```

### CODE REVIEW

File: jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java [hunk 1]
```diff
@@ -230,21 +234,76 @@ public static Password getPassword(String realm, String dft, String promptDft)
...
-            System.err.println("Usage - java " + Password.class.getName() + " [<user>] <password>");
-            System.err.println("If the password is ?, the user will be prompted for the password");
+            System.err.printf("Usage - java %s [<user>] <password> --prompt%n", Password.class.getName());
+            System.err.printf("Argument options:%n");
+            System.err.printf("  %s%n", Password.class.getName());
+            System.err.printf("     No arguments, will show this help%n");
+            System.err.printf("  %s myusername%n", Password.class.getName());
+            System.err.printf("     username only, will prompt for entries%n");
+            System.err.printf("  %s myusername ?%n", Password.class.getName());
+            System.err.printf("     username with question mark password, will prompt for entries%n");
```
Comment (gregw) [inline jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java]:
```suggestion
            System.err.printf("     username with question mark password, will prompt for arguments%n");
```

File: jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java [hunk 2]
```diff
@@ -230,21 +234,76 @@ public static Password getPassword(String realm, String dft, String promptDft)
...
+            System.err.printf("  %s myusername secretpassword%n", Password.class.getName());
+            System.err.printf("     username with password, will produce obfuscation results%n");
+            System.err.printf("  %s --prompt%n", Password.class.getName());
+            System.err.printf("     will prompt for entries%n");
```
Comment (gregw) [inline jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java]:
```suggestion
            System.err.printf("     will prompt for arguments%n");
```

**Review by gregw — DISMISSED** [review]
A niggle about the language used in the help page, but other than that it is OK

File: jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java [hunk 3]
```diff
@@ -230,21 +234,76 @@ public static Password getPassword(String realm, String dft, String promptDft)
...
+        }
+        else if (StringUtil.isBlank(argUser))
         {
-            System.err.println("Usage - java " + Password.class.getName() + " [<user>] <password>");
-            System.err.println("If the password is ?, the user will be prompted for the password");
+            System.err.printf("Usage - java %s [<user>] <password> --prompt%n", Password.class.getName());
+            System.err.printf("Argument options:%n");
+            System.err.printf("  %s%n", Password.class.getName());
+            System.err.printf("     No arguments, will show this help%n");
+            System.err.printf("  %s myusername%n", Password.class.getName());
```
Comment (janbartel) [inline jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java]:
```suggestion
            System.err.printf("  %s user%n", Password.class.getName());
```

File: jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java [hunk 4]
```diff
@@ -230,21 +234,76 @@ public static Password getPassword(String realm, String dft, String promptDft)
...
+            System.err.printf("     username only, will prompt for entries%n");
+            System.err.printf("  %s myusername ?%n", Password.class.getName());
```
Comment (janbartel) [inline jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java]:
```suggestion
            System.err.printf("  %s user ?%n", Password.class.getName());
```

File: jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java [hunk 5]
```diff
@@ -230,21 +234,76 @@ public static Password getPassword(String realm, String dft, String promptDft)
...
+            System.err.printf("     username with question mark password, will prompt for entries%n");
+            System.err.printf("  %s myusername secretpassword%n", Password.class.getName());
```
Comment (janbartel) [inline jetty-util/src/main/java/org/eclipse/jetty/util/security/Password.java]:
```suggestion
            System.err.printf("  %s user password%n", Password.class.getName());
```

**Review by janbartel — DISMISSED** [review]
I think the help text should be consistent. If the first line is `[<user>] <password>` then those terms should be used throughout.

**Review by sbordet — APPROVED** [review]

**Pull request merged**